### PR TITLE
Pop up modals on settings page with URL parameter

### PR DESCRIPTION
**Pull request by jessefriedman**
a requirement for Just In Time Messaging will be to send the user to a
place to learn about a module and have a call-to-action.

This will allow you to add “&info=module-slug” to the end of the
settings page URL and auto pop a modal.

### CODE REVIEW

File: class.jetpack-modules-list-table.php [hunk 1]
```diff
@@ -50,6 +50,7 @@ function __construct() {
 			'i18n'    => array(
 				'search_placeholder' => __( 'Search Modules…', 'jetpack' ),
 			),
+			'modalinfo' =>  isset( $_GET['info'] ) ? $_GET['info'] : false,
```
Comment (dereksmart) [inline class.jetpack-modules-list-table.php]: We should probably do some sanitization before passing this over.  Currently if I write something like `&info=flippideedoo` it'll pop a blank modal.

File: _inc/jetpack-modules.js [hunk 2]
```diff
@@ -59,6 +59,15 @@
 		$( '.modal' ).focus();
 	};
 
+	/**
+	 * If modalinfo is defined, auto popup the modal
+	 */
+	$( window ).load(function() {
```
Comment (zinigor) [inline _inc/jetpack-modules.js]: Are you sure we need to use `$( window ).load` here instead of `$( document ).ready`?

**Comment by zinigor** [comment]
Overall looks good, but needs some polishing. @jessefriedman if you need help with this one, please let me know.

File: _inc/jetpack-modules.js (hunk 2 above)
Comment (jessefriedman) [inline _inc/jetpack-modules.js]: @zinigor yes, it doesn't work with `$( document ).ready`. It has to do with the modal not being loaded yet at that point

**Comment by dereksmart** [comment]
latest changes look good, but now it needs a rebase

File: _inc/jetpack-modules.js (hunk 2 above)
Comment (jessefriedman) [inline _inc/jetpack-modules.js]: @zinigor I take it back it works fine with `$( document ).ready`

**Comment by jessefriedman** [comment]
closed in favor of #2270

**Pull request closed without merging**